### PR TITLE
Adds mocha test retry hook to reporter

### PR DESCRIPTION
**Pull request by RimantasDob**
## Proposed changes
Add extra reporter hook to be called if mocha test fails and will be rerun.
Currently, if test is set to have a retry and fails, no `wdio-reporter` hook is executed, leaving user without any knowledge why test fails. Error could be logged in wdio hook afterTest, but it is not called if test timeouts.

## Types of changes

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

I tested this for mocha only, not sure if it effects Jasmine or Cucumber users.

### CODE REVIEW

File: packages/wdio-reporter/tests/reporter.listeners.test.js [hunk 1]
```diff
@@ -540,6 +541,58 @@ describe('WDIOReporter Listeners', () => {
...
+            }
+            testEndEvent = {
+                uid: 'testid'
+            }
+        })
+
+        it('should call test retry with the test stat', () => {
+            reporter.onTestEnd = jest.fn()
+            emitEvents()
+            expect(reporter.onTestEnd).toHaveBeenCalledWith(reporter.tests[testStartEvent.uid])
```
Comment (christian-bromann) [inline packages/wdio-reporter/tests/reporter.listeners.test.js:592]: shouldn't this check `onTestRetry`? Also could we check if `reporter.retries` was changed too?

**Review by christian-bromann — CHANGES_REQUESTED** [review]
Thanks for the contribution! It would be good to strengthen the unit test a bit.

**Comment by christian-bromann** [comment]
Also would you mind to sign the CLA? Thanks!

File: packages/wdio-reporter/tests/reporter.listeners.test.js (hunk 1 above)
Comment (RimantasDob) [reply packages/wdio-reporter/tests/reporter.listeners.test.js]: Updated unit tests. Added tests for `retries` count, `passes` and `failures` count before and after retry.

**Comment by RimantasDob** [comment]
@christian-bromann I thought I signed CLA, didn't I.
Screenshot 2020-09-23 at 14 12 33

**Review by christian-bromann — APPROVED** [review]
Great addition, thanks 👍

**Pull request merged**